scan.go

```diff
@@ -221,12 +242,29 @@ func (s *ScanService) Do() (*ScanCursor, error) {
 		path += "/" + strings.Join(typesPart, ",")
 	}
 
+	// Define the body of the request
+	var body map[string]interface{}
+
+	if s.source != nil {
+		body = s.source
+	} else {
+		src, err := s.searchSource.Source()
+
+		if err != nil {
+			return nil, err
+		}
+
+		// searchSource.Source() actually returns a map, yet its return type
+		// is specified as interface...
```

Comment:
As an example, json.Unmarshal returns a very defined set of possible values in interface, bool, float64, str, nil, and []interface{} (an array of the aforementioned types or a map or an array). I can understand the use of interface{} there, because more than one type can be returned.